### PR TITLE
feat: add pt-br support

### PR DESCRIPTION
**Pull request by llldc21**
- Create a br page
- Update index to redirect for br page
- add engine errors in pt-br version
- updae parserError to accept pt-br version
- update tsErrorMessages to support pt-br translation

### CODE REVIEW

File: packages/engine/src/parseErrors.ts [hunk 1]
```diff
@@ -62,6 +68,7 @@ export interface ParseErrorsOpts {
 export const parseErrors = (
   message: string,
   opts?: ParseErrorsOpts,
+  language?: string,
```
Comment (mattpocock) [inline packages/engine/src/parseErrors.ts:71]: This should probably live in `opts`

**Comment by mattpocock** [comment]
Great PR! I'm concerned about the amount of duplication from the code that's already there, ideally we want to be able to support multiple languages without re-making the code each time. BUT I do like the errors being under a different `errors/lang` dir.

I'm going to let this sit while I think about how I want the code to look, and leave this PR open for discussion.

**Comment by miszo** [comment]
How about just using https://github.com/isaachinman/next-i18next for multiple languages? Together with https://github.com/i18next/i18next-scanner, you can scan for new strings that were added and needs translation.

**Comment by mattpocock** [comment]
@miszo I think a JSON file with the translations will probably be enough for the front page stuff. The app is unlikely to go to more than one page.

How would users access the translated pages? Via a dropdown on the homepage which takes them to `/[lang]`? Do we do anything clever with trying to work out their lang and redirect them there?

**Comment by miszo** [comment]
> @miszo I think a JSON file with the translations will probably be enough for the front page stuff. The app is unlikely to go to more than one page.

I mean, sure – the idea of keeping the TS errors is enough, but in the i18next, you could keep just “marketing” translations for the page. And the translations are also stored in JSON files.

> How would users access the translated pages? Via a dropdown on the homepage which takes them to `/[lang]`? Do we do anything clever with trying to work out their lang and redirect them there?

Why don't we have both? Auto-redirect can be done using Next.js locales. But I would leave an option to change language via some option on the page – depends on the number of languages. Because with 10+ languages, the dropdown wouldn't be the best idea 🙈 

Girl shrugs and says: “Why don't we have both?”

**Comment by mattpocock** [comment]
> I mean, sure – the idea of keeping the TS errors is enough, but in the i18next, you could keep just “marketing” translations for the page. And the translations are also stored in JSON files.

Yep, just nervous about adding tooling for tooling's sake. Translations in a single JSON file feels simplest.

This looks like it makes sense for us:

image

**Comment by miszo** [comment]
I entirely understand your point.

I just had a good experience with these tools and with them and wanted to recommend them, since they work well with Next.js.

**Comment by doiska** [comment]
i dont think we should translate the type itself, like:

`"A conversão do tipo '{0}' para o tipo '{1}' pode ser um erro porque nenhum tipo se sobrepõe suficientemente ao outro. Se isso foi intencional, converta a expressão para 'desconhecido' primeiro."`

should still be "unknown" instead of "desconhecido"

File: packages/engine/src/parseErrors.ts (hunk 1 above)
Comment (llldc21) [reply packages/engine/src/parseErrors.ts:71]: you ar right

**Comment by dohooo** [comment]
I will contribute to the Chinese translation hh

File: packages/engine/src/tsErrorMessages.json [hunk 2]
```diff
@@ -1717,7 +1717,10 @@
   },
   "Conversion of type '{0}' to type '{1}' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.": {
     "category": "Error",
-    "code": 2352
+    "code": 2352,
+    "translate": {
+      "br": "A conversão do tipo '{0}' para o tipo '{1}' pode ser um erro porque nenhum tipo se sobrepõe suficientemente ao outro. Se isso foi intencional, converta a expressão para 'desconhecido' primeiro."
```
Comment (RigottiG) [inline packages/engine/src/tsErrorMessages.json:1722]: I think should be `unknow` instead of `desconhecido`.

Comment (glaucia86) [reply packages/engine/src/tsErrorMessages.json:1722]: Types for sure and other keywords we shouldn't translated.

@mattpocock if you need a help someone to make a code review (in Portuguese) before to approve any new PR, I can make it.

p.s.: I work at Microsoft as a Developer Advocate in JavaScript.

**Pull request closed without merging**